services/motion/builtin/move_attempt.go

```diff
@@ -0,0 +1,81 @@
+package builtin
+
+import (
+	"context"
+	"sync"
+
+	goutils "go.viam.com/utils"
+
+	"go.viam.com/rdk/utils"
+)
+
+// moveResponse is a struct that is used to communicate the outcome of a moveAttempt.
+type moveResponse struct {
+	err     error
+	success bool
+}
+
+// moveAttempt is a struct whose lifetime lasts the duration of an attempt to complete a moveRequest
+// it contains a context in which the move call executes and tracks the goroutines that it spawns.
+type moveAttempt struct {
+	ctx               context.Context
+	cancelFn          context.CancelFunc
+	backgroundWorkers *sync.WaitGroup
+
+	request      *moveRequest
+	responseChan chan moveResponse
+}
+
+// newMoveAttempt instantiates a moveAttempt which can later be started.
+// The caller of this function is expected to also call the cancel function to clean up after instantiation.
+func newMoveAttempt(ctx context.Context, request *moveRequest) *moveAttempt {
+	cancelCtx, cancelFn := context.WithCancel(ctx)
+	var backgroundWorkers sync.WaitGroup
+
+	return &moveAttempt{
+		ctx:               cancelCtx,
+		cancelFn:          cancelFn,
+		backgroundWorkers: &backgroundWorkers,
+
+		request:      request,
+		responseChan: make(chan moveResponse, 1),
+	}
+}
+
+// start begins a new moveAttempt by using its moveRequest to create a plan, spawn relevant replanners, and finally execute the motion.
+// the caller of this function should monitor the moveAttempt's responseChan as well as the replanners' responseChan to get insight
+// into the status of the moveAttempt.
+func (ma *moveAttempt) start() {
+	plan, err := ma.request.plan(ma.ctx)
+	if err != nil {
+		ma.responseChan <- moveResponse{err: err}
```

Comment:
do we want to return if this happens? Otherwise, we will be polling for replanning when we never actually started executing a valid plan.

Reply:
That sounds good